releasenotes/notes/dynamics-requirement-6109a8846c0f88c2.yaml

```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    Qiskit Dynamics has been removed as a requirement and is now an optional developer-only dependency.
```

Comment:
Developer-only might be a little strong since it is also used for tutorials? I guess all users of qiskit-experiments are developers in some sense :slightly_smiling_face: